src/Servers/Kestrel/samples/PlaintextApp/Startup.cs

```diff
@@ -16,16 +19,34 @@ public class Startup
 
         public void Configure(IApplicationBuilder app)
         {
-            app.Run((httpContext) =>
+            app.Run(Plaintext);
+        }
+
+        private static Task Plaintext(HttpContext httpContext)
+        {
+            var payload = _helloWorldBytes;
+            var response = httpContext.Response;
+
+            response.StatusCode = 200;
+            response.ContentType = "text/plain";
+            response.ContentLength = payload.Length;
+
+            var vt = response.BodyPipe.WriteAsync(payload);
+            if (vt.IsCompletedSuccessfully)
+            {
+                // Signal consumption to the IValueTaskSource
+                vt.GetAwaiter().GetResult();
+                return Task.CompletedTask;
+            }
+            else
+            {
+                return AwaitResult(vt);
```

Comment:
Is this better than `return vt.AsTask()`?

Reply:
See the thread above.

Reply:
Just saw that. I still don't see how `AsTask()` returning a `Task<FlushResult>` proves it's less efficient.

Reply:
It'll have to allocate in completed case.

Reply:
If we replace the entire  `if (vt.IsCompletedSuccessfully) {} else {} ` check with `vt.AsTask()`

Reply:
That'd be awesome if we could replace the whole if/else with AsTask(), but that has nothing to do with whether the current AsTask() is better than AwaitResult().

Reply:
I agree that we should prefer AsTask until proven otherwise.

Reply:
Don't use AsTask, it'll allocate.

Reply:
AwaitResult allocates as well.

Reply:
True, it might be the same. It is the slow path so maybe it doesn't matter as much.